skimage/morphology/misc.py

```diff
@@ -235,3 +243,208 @@ def remove_small_holes(ar, area_threshold=64, connectivity=1, *, out=None):
     np.logical_not(out, out=out)
 
     return out
+
+
+def remove_near_objects(
+    label_image,
+    min_distance,
+    *,
+    priority=None,
+    p_norm=2,
+    spacing=None,
+    out=None,
+):
+    """Remove objects until a minimal distance is ensured.
+
+    Iterates over all objects (pixels that aren't zero) inside an image and
+    removes "nearby" objects until all remaining ones are spaced more than a
+    given minimal distance from each other.
+
+    Parameters
+    ----------
+    label_image : ndarray of integers
+        An n-dimensional array containing object labels, e.g. as returned by
+        :func:`~.label`. A value of zero is considered background, all other
```

Comment:
As an aside, have you looked into the working of label? It builds a graph of the graph, and creates unions of those graphs. It sounds a lot like what you are doing in the internal function above.